test/Integration.t.sol

```diff
@@ -0,0 +1,311 @@
+            uint256 newGap = preIncreaseSupplyGap * 2;
+
+            vm.prank(SPARK_PROXY);
+            capAutomator.setSupplyCapConfig({
+                asset:            assets[i],
+                max:              newCap,
+                gap:              newGap,
+                increaseCooldown: 12 hours
+            });
+
+            capAutomator.exec(assets[i]);
+
+            uint256 postIncreaseSupplyCap = pool.getReserveData(assets[i]).configuration.getSupplyCap();
+
+            assertEq(postIncreaseSupplyCap, currentSupply + newGap);
+        }
+    }
+
+    function test_E2E_decreaseSupply() public {
+        for(uint256 i; i < assets.length; i++) {
+            DataTypes.ReserveData memory reserveData = pool.getReserveData(assets[i]);
+
+            uint256 preDecreaseSupplyCap = reserveData.configuration.getSupplyCap();
+            if (preDecreaseSupplyCap == 0) {
+                continue;
+            }
+
+            uint256 currentSupply = (IScaledBalanceToken(reserveData.aTokenAddress).scaledTotalSupply() + uint256(reserveData.accruedToTreasury)).rayMul(reserveData.liquidityIndex)
+                / 10 ** ERC20(reserveData.aTokenAddress).decimals();
+
+            uint256 preDecreaseSupplyGap = preDecreaseSupplyCap - currentSupply;
+
+            uint256 newGap = preDecreaseSupplyGap / 2;
+
+            vm.prank(SPARK_PROXY);
+            capAutomator.setSupplyCapConfig({
+                asset:            assets[i],
+                max:              preDecreaseSupplyCap,
+                gap:              newGap,
+                increaseCooldown: 12 hours
+            });
+
+            capAutomator.exec(assets[i]);
+
+            uint256 postDecreaseSupplyCap = pool.getReserveData(assets[i]).configuration.getSupplyCap();
+
+            assertEq(postDecreaseSupplyCap, currentSupply + newGap);
+        }
+    }
+
+}
+
+contract ConcreteTests is CapAutomatorIntegrationTestsBase {
+
+    using ReserveConfiguration for DataTypes.ReserveConfigurationMap;
+    using WadRayMath           for uint256;
+
+    function test_E2E() public {
+        assertEq(ERC20(WBTC).decimals(), 8);
+
+        DataTypes.ReserveData memory wbtcReserveData = pool.getReserveData(WBTC);
+
+        // Confirm initial supply cap
+        assertEq(wbtcReserveData.configuration.getSupplyCap(), 3_000);
+
+        // Confirm initial WBTC supply
+        uint256 initialSupply = (IScaledBalanceToken(wbtcReserveData.aTokenAddress).scaledTotalSupply() + uint256(wbtcReserveData.accruedToTreasury)).rayMul(wbtcReserveData.liquidityIndex)
+            / 10 ** ERC20(wbtcReserveData.aTokenAddress).decimals();
+        assertEq(initialSupply, 750);
+
+        vm.prank(SPARK_PROXY);
+        capAutomator.setSupplyCapConfig({
+            asset:            WBTC,
+            max:              6_000,
+            gap:              500,
+            increaseCooldown: 12 hours
+        });
+
+        uint256 stash = 6_000e8;
```

Comment:
What is this used for?

Reply:
The amount of tokens given to the user for deposits. Value in a variable, because it's used twice - in deal and in approve.